shop/controllers/product.py

```diff
@@ -0,0 +1,13 @@
+from shop.dal.product import ProductDAL
+
+
+class ProductController:
+    @classmethod
+    def get_product_list(cls, category_pk):
+        products = ProductDAL.get_all_or_category_products(category_pk)
+        return products
+
+    @classmethod
+    def get_product(cls, product_pk):
+        product = ProductDAL.get_product_by_pk(product_pk)
```

Comment:
where is catch the expection?

Reply:
It is inside of `get_product_by_pk`

Reply:
we need to separate layer that requests data and logic layer.